cpp/src/arrow/filesystem/gcsfs.cc

```diff
@@ -659,8 +715,8 @@ Result<std::shared_ptr<io::InputStream>> GcsFileSystem::OpenInputStream(
 
 Result<std::shared_ptr<io::InputStream>> GcsFileSystem::OpenInputStream(
     const FileInfo& info) {
-  if (!info.IsFile()) {
-    return Status::IOError("Only files can be opened as input streams");
+  if (info.IsDirectory()) {
+    return Status::IOError("Cannot open a directory as an input stream");
```

Comment:
maybe include the path requested here?

Reply:
Fixed.  I also searched for any other places where I was not returning the affected path with the IOError.